### ISSUE TITLE
ordered-imports not working as expected with dots in file name

### ISSUE THREAD

**Issue by fallXone** [palantir/tslint#2484]
### Bug Report

- __TSLint version__: 3.10.3
- __TypeScript version__: 2.2.1
- __Running TSLint via__: (pick one)  VSCode 

#### TypeScript code being linted

```ts
import { ICS } from '../../../contracts/c-s.interface';
import { ICSDV } from '../../../contracts/c-s-d-v.interface';
import { ICSGV } from '../../../contracts/c-s-g-v.interface';
```

with `tslint.json` configuration:

```json
"ordered-imports": [
            true
        ],
```

#### Actual behavior
```ts
import { ICSDV } from '../../../contracts/c-s-d-v.interface';
import { ICSGV } from '../../../contracts/c-s-g-v.interface';
import { ICS } from '../../../contracts/c-s.interface';
```
#### Expected behavior
```ts
import { ICS } from '../../../contracts/c-s.interface';
import { ICSDV } from '../../../contracts/c-s-d-v.interface';
import { ICSGV } from '../../../contracts/c-s-g-v.interface';
```

**Comment by adidahiya** [comment]
[Here](https://github.com/palantir/tslint/blob/bbb0fe9a1c5eb547467533f2750ac2799efec214/src/rules/orderedImportsRule.ts#L108) is the code that compares strings in ordered-imports. We're using JavaScript's built-in [standard lexicographical ordering](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators?v=control) of strings. Do you have any reference you can link to that supports your proposed change in the expected behavior? Or is it a personal subjective ordering preference?

**Comment by JoshuaKGoldberg** [comment]
Closing for housekeeping - @fallXone, if you have a proposal for how this could be changed, please feel free to post it here!
